### PR TITLE
#164798911 Implement create article endpoint

### PR DESCRIPTION
**Pull request by geekdanitek**
#### Type of change
- [x] Endpoint to create article
- [x] Write unit testing for create article endpoint
- [x] Document endpoint using swagger
#### How Has This Been Tested?
- All unit test passed
-  Successfully created new article for user using postman
Screenshot 2019-04-05 at 10 44 37 AM

#### Checklist:
- [x] Implement endpoint to create new article
- [x] All test passed
- [x] Documented endpoint
#### Story (PT)
[Delivers #164798913](https://www.pivotaltracker.com/story/show/164798913)

### CODE REVIEW

**Comment by TheDrizzyWay** [comment]
Good work. you have two commits though #pair-programming

**Comment by Chiazokam** [comment]
Looks good. Great work!

File: server/models/article.js [hunk 1]
```diff
@@ -7,28 +7,39 @@ module.exports = (sequelize, DataTypes) => {
...
+      type: DataTypes.UUID
     },
   }, {});
-  Article.associate = (models) => {};
+  Article.associate = (models) => {
+    Article.belongsTo(models.User, {
+      foreignkey: 'userId',
+      onDelete: 'CASCADE'
+    });
+  };
```
Comment (samorahno) [inline server/models/article.js:43]: Good Job

File: server/test/article.spec.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+      const res = await chai.request(app)
+        .post('/api/v1/article')
+        .set({ Authorization: `Bearer ${userToken}` })
+        .send(articleMockData.invalidObject[11]);
+      const { message } = res.body;
+      expect(res).to.have.status(422);
+      expect(message).to.be.equal('title, body, description, tags and published fields are required');
+    });
+  });
+});
```
Comment (samorahno) [inline server/test/article.spec.js:171]: Good job. Detailed test

**Review by samorahno — COMMENTED** [review]
Good Job. Well done

File: server/controllers/article.controller.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      slug: slug(`${title} ${Date.now()}`),
+      userId: id
+    });
+    if (!article) {
+      return Response(res, 400, errorMessage);
+    }
+    return Response(res, 201, successMessage, { article });
+  }
+}
+export default ArticleController;
```
Comment (davidokonji) [inline server/controllers/article.controller.js:42]: Very neat work, good one guys

**Comment by geekdanitek** [comment]
> Good work. you have two commits though #pair-programming
@TheDrizzyWay Yeah will squash all commit into one

**Comment by paradisekelechi** [comment]
@geekdanitek Is `#164798911-story(api) Implement create article endpoint` proper naming convention?

**Comment by geekdanitek** [comment]
> @geekdanitek Is `#164798911-story(api) Implement create article endpoint` proper naming convention?

@paradisekelechi Sorry for the mistake will change it now.

File: server/utils/isRequired.util.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * @desciption - Specify a key of an array that is required in another array
+ *
+ * @param {string} haystack
+ * @param {string} needle
+ * @returns {string} slug
+ */
+const isRequired = (haystack, needle) => {
+  const required = [];
+  // eslint-disable-next-line array-callback-return
```
Comment (paradisekelechi) [inline server/utils/isRequired.util.js]: Check out this resource https://eslint.org/docs/rules/array-callback-return and revert accordingly

**Review by paradisekelechi — CHANGES_REQUESTED** [review]

File: server/utils/isRequired.util.js (hunk 4 above)
Comment (geekdanitek) [reply server/utils/isRequired.util.js]: Ok, will implement now.

File: server/test/article.spec.js (hunk 2 above)
Comment (BukkyOmo) [reply server/test/article.spec.js:171]: Thank you!

File: server/utils/isRequired.util.js (hunk 4 above)
Comment (geekdanitek) [reply server/utils/isRequired.util.js]: This has been implemented.

**Pull request merged**